stackable-operator-derive/src/lib.rs

```diff
@@ -100,127 +50,90 @@ enum InputType {
-                };
-                quote! {
-                    #pattern => {#body},
-                }
-            },
-        )
-        .collect::<TokenStream>();
-
-    if let Some(bounds) = bounds {
-        let where_clause = generics.make_where_clause();
-        where_clause.predicates.extend(bounds);
-    }
-    let (impl_generics, ty_generics, where_clause) = generics.split_for_impl();
-    let ty_toks = match ty {
-        InputType::Struct => quote! { #ident #ty_generics },
-        // Enums need some way to indicate that we want to keep the same variant, in our case we use
-        // Option::None to signal this
-        InputType::Enum => quote! { Option<#ident #ty_generics> },
-    };
-    let fallback_variants = match ty {
-        InputType::Struct => quote! {},
-        InputType::Enum => quote! {
-            // self is None => inherit everything from defaults
-            (this @ None, defaults) => *this = <Self as ::std::clone::Clone>::clone(defaults),
-            // self is Some but mismatches defaults, discard defaults
-            (Some(_), _) => {}
-        },
-    };
-    quote! {
-        impl #impl_generics #merge_mod::Merge for #ty_toks #where_clause {
-            fn merge(&mut self, defaults: &Self) {
-                match (self, defaults) {
-                    #merge_variants
-                    #fallback_variants
-                }
-            }
-        }
-    }
-    .into()
-}
-
-fn map_fields_to_prefixed_vars<'a>(
-    constructor: &Path,
-    fields: impl IntoIterator<Item = Option<&'a Ident>>,
-    prefix: &Ident,
-) -> TokenStream {
-    let fields = fields
-        .into_iter()
-        .enumerate()
-        .map(|(index, field)| {
-            let prefixed = prefix_ident(field.ok_or(index), prefix);
-            if let Some(field) = field {
-                quote! { #field: #prefixed, }
-            } else {
-                let index = Index::from(index);
-                quote! { #index: #prefixed, }
-            }
-        })
-        .collect::<TokenStream>();
-    quote! { #constructor { #fields } }
+    merge::derive(parse_macro_input!(input)).into()
 }
 
-fn prefix_ident(ident: Result<&Ident, usize>, prefix: &Ident) -> Ident {
-    match ident {
-        Ok(ident) => format_ident!("{prefix}_{ident}"),
-        Err(index) => format_ident!("{prefix}_{index}"),
-    }
+/// Creates a [fragment type](index.html) for the given type, and implements [`FromFragment`](trait.FromFragment.html).
+///
+/// This macro implements "deep optionality", meaning that each field is replaced by its `Fragment` variant. For example, this type:
+///
+/// ```
+/// #[derive(Fragment)]
+/// struct Foo {
+///     atomic: u32,
+///     nested: Bar,
+/// }
+/// ```
+///
+/// would generate the following output:
+///
+/// ```
+/// struct FooFragment {
+///     atomic: u32::Fragment, // = Option<u32>
+///     nested: Bar::Fragment, // = BarFragment, assuming that Bar is also #[derive(Fragment)]
```

Comment:
If `Bar` implements `FromFragment` manually then it will be whatever that impl defines `Bar::Fragment` to be. Otherwise the user will get a compile error complaining about `FromFragment` being required.

Reply:
Thanks for the clarification!